tedana/decay.py

```diff
@@ -11,6 +11,35 @@
 RefLGR = logging.getLogger('REFERENCES')
 
 
+def _apply_t2s_floor(t2s, echo_times):
+    """
+    Apply a floor to T2* values to prevent zero division errors during
+    optimal combination.
+
+    Parameters
+    ----------
+    t2s : (S,) array_like
+        T2* estimates.
+    echo_times : (E,) array_like
+        Echo times in milliseconds.
+
+    Returns
+    -------
+    t2s_corrected : (S,) array_like
+        T2* estimates with very small, positive values replaced with a floor value.
+    """
+    t2s_corrected = t2s.copy()
+    echo_times = np.asarray(echo_times)
+    if echo_times.ndim == 1:
+        echo_times = echo_times[:, None]
+
+    eps = np.finfo(dtype=t2s.dtype).eps  # smallest value for datatype
+    temp_arr = np.exp(-echo_times / t2s)  # (E x V) array
+    bad_voxel_idx = np.any(temp_arr == 0, axis=0) & (t2s != 0)
```

Comment:
Is the pythonic way considered to be using `and` instead?

Reply:
I've been silly and forgot about bitwise operations, ignore me.

Reply:
`and` doesn't do element wise comparisons like that. Replacing `&` with `and` in this case gets the following error:

```python
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all() 
```